Source/WebGPU/WebGPU/Buffer.mm

```diff
@@ -356,6 +356,22 @@ void wgpuBufferDestroy(WGPUBuffer buffer)
     return WebGPU::fromAPI(buffer).getConstMappedRange(offset, size);
 }
 
+WGPUBufferMapState wgpuBufferGetMapState(WGPUBuffer buffer)
+{
+    switch (WebGPU::fromAPI(buffer).state()) {
+    case WebGPU::Buffer::State::Mapped:
```

Comment:
Should we just get rid of our own enum in favor of the WGPU one?

Reply:
Our own enum has additional information (MappedAtCreation and Destroy), I'm not sure if that is useful.